src/s1reader/s1_reader.py

```diff
@@ -860,9 +850,106 @@ def _burst_from_safe_dir(safe_dir_path: str, id_str: str, orbit_path: str, flag_
     else:
         msg = f'measurement directory NOT found in {safe_dir_path}'
         msg += ', continue with metadata only.'
-        print(msg)
+        # print(msg)
         f_tiff = ''
 
     bursts = burst_from_xml(f_annotation, orbit_path, f_tiff, iw2_f_annotation,
                             flag_apply_eap=flag_apply_eap)
     return bursts
+
+
+def get_burst_id(
+    sensing_time: datetime.datetime,
+    ascending_node_dt: datetime.datetime,
+    start_track: int,
+    end_track: int,
+    subswath_name: str,
+) -> int:
+    """Calculate burst ID and current track number of a burst.
+
+    Accounts for equator crossing frames, and uses the ESA convention defined
+    in the Sentinel-1 Level 1 Detailed Algorithm Definition
+
+    Parameters
+    ----------
+    sensing_time : datetime
+        Sensing time of the first input line of this burst [UTC]
+        The XML tag is sensingTime in the annotation file.
+    ascending_node_dt : datetime
+        Time of the ascending node prior to the start of the scene.
+    start_track : int
+        Relative orbit number at the start of the acquisition, from 1-175.
+    end_track : int
+        Relative orbit number at the end of the acquisition.
+    subswath_name : str, {'IW1', 'IW2', 'IW3'}
+        Name of the subswath of the burst.
+
+    Returns
+    -------
+    relative_orbit : int
+        Relative orbit number (track number) at the current burst.
+    burst_id : int
+        The burst ID matching ESA's relative numbering scheme.
+
+    Notes
+    -----
+    The `start_track` and `end_track` parameters are used to determine if the
+    scene crosses the equator. They are the same if the frame does not cross
+    the equator.
+
+    References
+    ----------
+    ESA Sentinel-1 Level 1 Detailed Algorithm Definition
+    https://sentinels.copernicus.eu/documents/247904/1877131/S1-TN-MDA-52-7445_Sentinel-1+Level+1+Detailed+Algorithm+Definition_v2-4.pdf/83624863-6429-cfb8-2371-5c5ca82907b8
+    """
+    # Constants in Table 9-7
```

Comment:
Wondering again whether we should have a separate file in the repository where to insert all the constants

Reply:
Same comment here- I can move these to a constants file if you'd like, but i'm unsure if the constants would be easier to interpret if they're moved away from function that uses them. If we're planning on making a big `constants.py` then I can start with these

Reply:
I don't think there's enough for a dedicated constants file. From a quick glance, I didn't see anything beyond 3 constants below in the repo. 

What do you about moving `get_burst_id` into its own file? A `s1_burst_id.py` that's something like the following (some inspiration [from](https://stackoverflow.com/questions/61937520/proper-way-to-create-class-variable-in-data-class)):
```python
from typing import ClassVar
from dataclasses import dataclass

@dataclass
class S1BurstID:
   T_beam: ClassVar[float] = 2.758273  # interval of one burst [s]
    T_pre: ClassVar[float] = 2.299849  # Preamble time interval [s]
    T_orb: ClassVar[float] = 12 * 24 * 3600 / 175  # Nominal orbit period [s]
    track_number: int
    esa_burst_id: int
    subswath_name: str

    @classmethod
    def params_to_id(cls, 
        sensing_time: datetime.datetime, 
        ascending_node_dt: datetime.datetime,
        start_track: int,
        end_track: int,
        subswath_name: str):
        # do same computations
        return cls(track_numer, esa_burst_id, subswath)

    @property
    def as_str(self):
        return f"t{self.track_number:03d}_{self.esa_burst_id:06d}_{self.subswath_name.lower()}"
```

We can then change [this](https://github.com/opera-adt/s1-reader/blob/cf857643af13a811aaa7c5f4ce21d54f0f093290/src/s1reader/s1_burst_slc.py#L147) from `str` to `S1BurstID`

By importing this module, all the constants as accessible a class properties.